### PR TITLE
PC-1313: Better messaging for routing errors

### PR DESCRIPTION
**Pull request by samyou-softwire**
[Link to Jira ticket](https://beisdigital.atlassian.net/browse/PC-1313)

# Description

adds a new page for an error calculating the users' journey

the case that throws this error is now checked explicitly (rather than relying on back button routing error)

the exceptions to this rule have been named more appropriately

translations have been added

# Checklist

- [x] I have made any necessary updates to the documentation
- [x] I have checked there are no unnecessary IDE warnings in this PR
- [x] I have checked there are no unintentional line ending changes
- [x] I have added tests where applicable
- [x] If I have made any changes to the code, I have used the IDE auto-formatter on it
- [x] If I have made any changes to Python files (even if not changing any content strings), I have run `make extract-translations`
- [x] If I have made any changes to website flow, I have updated the [Flow Miro Board](https://miro.com/app/board/uXjVK4AgQ3E=/)
- [x] If I have made any changes to website flow, I have checked forward and back behaviour is still consistent

# Screenshots

image
image

### CODE REVIEW

**Review by Glenn-Clarke — APPROVED** [review]
LGTM

File: help_to_heat/frontdoor/views.py [hunk 1]
```diff
@@ -371,6 +375,13 @@ def get(self, request, session_id, page_name, unsaved_data=None, errors=None, is
         if data_with_get_answers != data:
             save_answer(session_id, page_name, data_with_get_answers)
 
+        if page_name not in schemas.routing_overrides:
+            try:
+                # ensure the user has answers to complete the journey to this page
+                calculate_journey(answers, to_page=page_name)
+            except CouldNotCalculateJourneyException:
+                return redirect("/sorry-journey")
```
Comment (jdgage) [inline help_to_heat/frontdoor/views.py:383]: Do we want to log a warning or something else below an error here now? I forget whether that's covered by another ticket.

Comment (samyou-softwire) [reply help_to_heat/frontdoor/views.py:383]: good point; we do get the choice here for whether we want to or not

I'd err on doing so for now (I'll update the code), then we'll monitor going forwards whether we still want to or not (covered by that other ticket)

File: help_to_heat/frontdoor/views.py [hunk 2]
```diff
@@ -371,6 +375,14 @@ def get(self, request, session_id, page_name, unsaved_data=None, errors=None, is
         if data_with_get_answers != data:
             save_answer(session_id, page_name, data_with_get_answers)
 
+        if page_name not in schemas.routing_overrides:
+            try:
+                # ensure the user has answers to complete the journey to this page
+                calculate_journey(answers, to_page=page_name)
+            except CouldNotCalculateJourneyException as e:
+                logger.exception(e)
```
Comment (jdgage) [inline help_to_heat/frontdoor/views.py:383]: Ok, so to confirm, this is the equivalent to what we currently do, we just send them to a different page?

Comment (samyou-softwire) [reply help_to_heat/frontdoor/views.py:383]: yes, to be specific:
- currently, if you try to view a page that's not on your route, you will see an error page
- this happens when trying to calculate the previous page
- calculating the previous page requires finding a route to your page, and going 1 back
- the code will fail on trying to find a route to your page
- this has been moved to an explicit check
- so functionality should be identical to before

Comment (jdgage) [reply help_to_heat/frontdoor/views.py:383]: Yes, and in particular we'll still get an error logged for it?

**Review by jdgage — APPROVED** [review]

File: help_to_heat/frontdoor/views.py (hunk 2 above)
Comment (samyou-softwire) [reply help_to_heat/frontdoor/views.py:383]: yes (the one in this diff)

**Pull request merged**